File: frontend/src/api/scenario.ts
import { http } from '@/utils/requests'
import type { ApiResponse } from '@/types/api'
import type {
  QueryScenarioList,
  QueryScenarioOne,
  ScenarioCreate,
  ScenarioInfo,
  ScenarioList,
  ScenarioUpdate,
} from '@/types/scenario'

export const ScenarioApi = {
  getScenarioList(data: QueryScenarioList): Promise<ApiResponse<ScenarioList>> {
    return http.post<ApiResponse<ScenarioList>>('/api/scenario/list', data)
  },

  getScenarioInfo(data: QueryScenarioOne): Promise<ApiResponse<ScenarioInfo>> {
    return http.post<ApiResponse<ScenarioInfo>>('/api/scenario/info', data)
  },

  createScenario(data: ScenarioCreate): Promise<ApiResponse<ScenarioInfo>> {
    return http.post<ApiResponse<ScenarioInfo>>('/api/scenario/create', data)
  },

  updateScenario(data: ScenarioUpdate): Promise<ApiResponse<ScenarioInfo>> {
    return http.post<ApiResponse<ScenarioInfo>>('/api/scenario/update', data)
  },

  deleteScenario(data: QueryScenarioOne): Promise<ApiResponse<null>> {
    return http.post<ApiResponse<null>>('/api/scenario/delete', data)
  },
}

File: falcon-master/app/api/scenario.py
# falcon_shared/api/v1/scenario.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session

from app.core.response import SuccessResponse
from app.db import get_db
from app.schemas import scenario as schemas
from app.services.scenario_service import ScenarioService

router = APIRouter(prefix="/scenario", tags=["Scenario"])


@router.post("/list", response_model=schemas.ScenarioListResponse)
async def scenario_list(data:schemas.QueryScenarioList, db: Session = Depends(get_db),):


    scenario_service = ScenarioService(db)

    scenarios = scenario_service.list(**data.model_dump(mode="json"))

    return SuccessResponse(data=scenarios)

@router.post("/info", response_model=schemas.ScenarioInfoResponse)
async def scenario_info(data: schemas.QueryScenarioOne, db: Session = Depends(get_db),):

    scenario_service = ScenarioService(db)

    scenarios = scenario_service.get(**data.model_dump())

    return SuccessResponse(data=scenarios)


@router.post("/delete", response_model=schemas.BaseResponse)
async def scenario_delete(data: schemas.QueryScenarioOne, db: Session = Depends(get_db)):

    scenario_service = ScenarioService(db)

    scenario_service.delete(**data.model_dump())

    return SuccessResponse(data=None)

@router.post("/create", response_model=schemas.ScenarioInfoResponse)
async def scenario_create(data: schemas.ScenarioCreate, db: Session = Depends(get_db)):

    scenario_service = ScenarioService(db)

    scenarios = scenario_service.create(data)

    return SuccessResponse(data=scenarios)

@router.post("/update", response_model=schemas.ScenarioInfoResponse)
async def scenario_update(data: schemas.ScenarioUpdate, db: Session = Depends(get_db)):

    scenario_service = ScenarioService(db)

    scenarios = scenario_service.update(data)

    return SuccessResponse(data=scenarios)
